src/sentry/preprod/vcs/status_checks/tasks.py

```diff
@@ -0,0 +1,326 @@
+from sentry.integrations.models.integration import Integration
+from sentry.integrations.source_code_management.metrics import (
+    SCMIntegrationInteractionEvent,
+    SCMIntegrationInteractionType,
+)
+from sentry.integrations.source_code_management.status_check import (
+    StatusCheckClient,
+    StatusCheckStatus,
+)
+from sentry.integrations.types import IntegrationProviderSlug
+from sentry.models.commitcomparison import CommitComparison
+from sentry.models.project import Project
+from sentry.models.repository import Repository
+from sentry.preprod.models import PreprodArtifact
+from sentry.silo.base import SiloMode
+from sentry.tasks.base import instrumented_task
+from sentry.taskworker.config import TaskworkerConfig
+from sentry.taskworker.namespaces import integrations_tasks
+from sentry.taskworker.retry import Retry
+
+logger = logging.getLogger(__name__)
+
+SIZE_ANALYZER_TITLE = "Sentry Size Analysis"  # TODO(preprod): translate
+
+
+@instrumented_task(
+    name="sentry.preprod.tasks.create_preprod_status_check",
+    queue="integrations",
+    silo_mode=SiloMode.REGION,
+    retry=Retry(times=3),
+    taskworker_config=TaskworkerConfig(
+        namespace=integrations_tasks,
+        processing_deadline_duration=30,
+    ),
+)
+def create_preprod_status_check_task(preprod_artifact_id: int) -> None:
+    try:
+        preprod_artifact = PreprodArtifact.objects.get(id=preprod_artifact_id)
+    except PreprodArtifact.DoesNotExist:
+        logger.exception(
+            "preprod.status_checks.create.artifact_not_found",
+            extra={"artifact_id": preprod_artifact_id},
+        )
+        return
+
+    logger.info(
+        "preprod.status_checks.create.start",
+        extra={"artifact_id": preprod_artifact.id},
+    )
+
+    title = SIZE_ANALYZER_TITLE
+
+    # TODO(preprod): add real formatting
+    if (
+        preprod_artifact.state == PreprodArtifact.ArtifactState.UPLOADING
+        or preprod_artifact.state == PreprodArtifact.ArtifactState.UPLOADED
+    ):
+        status = StatusCheckStatus.IN_PROGRESS
+        text = f"⏳ Build {preprod_artifact.id} for {preprod_artifact.app_id} is processing..."
+        summary = f"Build {preprod_artifact.id} for `{preprod_artifact.app_id}` is processing..."
+        target_url = None
+    elif preprod_artifact.state == PreprodArtifact.ArtifactState.FAILED:
+        status = StatusCheckStatus.FAILURE
+        text = f"❌ Build {preprod_artifact.id} failed. Error: {preprod_artifact.error_message}"
+        summary = f"Build {preprod_artifact.id} for `{preprod_artifact.app_id}` failed."
+        target_url = None
+    elif preprod_artifact.state == PreprodArtifact.ArtifactState.PROCESSED:
+        status = StatusCheckStatus.SUCCESS
+        text = f"✅ Build {preprod_artifact.id} processed successfully."
+        summary = (
+            f"Build {preprod_artifact.id} for `{preprod_artifact.app_id}` processed successfully."
+        )
+        target_url = None
```

Comment:
should we set the target_url here?

Reply:
I wasn't sure how to get a URL that works across prod/devboxes, was going to revisit when I implement the more detailed formatting.